tests/setup_teardown.rs

```diff
@@ -26,130 +46,334 @@ pub async fn get_index(user: &GooseUser) -> GooseTaskResult {
     Ok(())
 }
 
+// All tests in this file run against common endpoints.
+fn setup_mock_server_endpoints(server: &MockServer) -> Vec<MockRef> {
+    let mut endpoints: Vec<MockRef> = Vec::new();
+
+    // First set up INDEX_PATH, store in vector at INDEX_KEY.
+    endpoints.push(
+        Mock::new()
+            .expect_method(GET)
+            .expect_path(INDEX_PATH)
+            .return_status(201)
+            .create_on(&server),
+    );
+    // Next set up SETUP_PATH, store in vector at SETUP_KEY.
+    endpoints.push(
+        Mock::new()
+            .expect_method(POST)
+            .expect_path(SETUP_PATH)
+            .return_status(205)
+            .create_on(&server),
+    );
+    // Next set up TEARDOWN_PATH, store in vector at TEARDOWN_KEY.
+    endpoints.push(
+        Mock::new()
+            .expect_method(POST)
+            .expect_path(TEARDOWN_PATH)
+            .return_status(200)
+            .create_on(&server),
+    );
+
+    endpoints
+}
+
+// Create a custom configuration for this test.
+fn common_build_configuration(
+    server: &MockServer,
+    worker: Option<bool>,
+    manager: Option<usize>,
+) -> GooseConfiguration {
+    if let Some(expect_workers) = manager {
+        common::build_configuration(
+            &server,
+            vec![
+                "--manager",
+                "--expect-workers",
+                &expect_workers.to_string(),
+                "--users",
+                USERS,
+                "--hatch-rate",
+                USERS,
+            ],
+        )
+    } else if worker.is_some() {
+        common::build_configuration(&server, vec!["--worker"])
+    } else {
+        common::build_configuration(&server, vec!["--users", USERS, "--hatch-rate", USERS])
+    }
+}
+
+// Common validation for the load tests in this file.
+// @FIXME: remove the `is_gaggle` flag once issue #182 lands.
+fn validate_test(test_type: &TestType, mock_endpoints: &Vec<MockRef>, is_gaggle: bool) {
+    // Confirm the load test ran.
+    assert!(mock_endpoints[INDEX_KEY].times_called() > 0);
+
+    // Now confirm TestType-specific counters.
+    if is_gaggle {
+        // @FIXME: when https://github.com/tag1consulting/goose/issues/182 lands
```

Comment:
The `is_gaggle` flag will need to be removed when #182 lands (and was setup so it will start failing when Gaggle's handling of `test_start` and `test_stop` are fixed so we don't forget).